common/src/paused_program_storage.rs

```diff
@@ -92,15 +84,51 @@ pub struct ResumeResult<BlockNumber> {
     pub info: Option<(ProgramId, BlockNumber)>,
 }
 
+/// The enumeration defines result of the [`PausedProgramStorage::pause_program()`] method.
+pub enum PauseResult {
+    /// Pausing a program has been started. The variant contains
+    /// the corresponding map with gas reservations.
+    Started(GasReservationMap),
+    /// The current step of pausing a program has been finished.
+    /// The variant contains the number of hashed memory pages.
+    InProcess(u32),
+    /// Pausing a program has been finished.
+    Finished,
+    /// Program has been paused in one step. The variant contains
+    /// the corresponding map with gas reservations.
+    Paused(GasReservationMap, u32),
+    /// Program is not initialized. The variant contains the id
+    /// of the corresponding init message.
+    Uninitialized(MessageId),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Decode, Encode, TypeInfo)]
+#[codec(crate = codec)]
+#[scale_info(crate = scale_info)]
+pub struct PausingProgramInfo<BlockNumber> {
```

Comment:
You're absolutely right! To tell the truth I titled it intentionally that way since `BeingPaused...` is a bit longer name. In any case this is an internal data struct and shouldn't be used outside.

Nonetheless if you insist I will rename it to `BeingPausedProgramInfo`.

Reply:
For me it will be much better